lib.coffee

```diff
@@ -308,8 +308,18 @@ class globals.Document
 
       fieldsWalker @meta.fields
 
-      referencesIncludeProjection = LocalCollection._compileProjection referencesInclude
       referencesExcludeProjection = LocalCollection._compileProjection referencesExclude
+      referencesIncludeModification = (document) ->
+        # We can't use _compileProjection here as this would cause top-level fields to
+        # be overwritten when updating subfields.
+        result = {}
+        for path, value of referencesInclude
+          continue if value isnt 1
+
+          value = _.reduce path.split('.'), ((doc, atom) -> doc[atom]), document
+          result[path] = value unless _.isUndefined value
```

Comment:
> In this case, the proper solution would be to set the field to [] instead of null?

Or simply omit the field. Not trying to set it to `null` at all. But yea, we can also be explicit. The point is, if source document didn't contain an optional field (author does not want to even set it), with current code it seems we force it to `null`, no? So I would agree that optional reference fields should be `null` in the first pass, but only if they really exist and will be set in the second pass. If the do not exist at all in the first pass, they should not be set by our code.